### PR TITLE
:sparkles: Add missing fields (WIP)

### PR DESCRIPTION
**Pull request by evans-g-crsj**

### CODE REVIEW

File: import-task/src/main/scala/bio/ferlab/fhir/etl/Utils.scala [hunk 1]
```diff
@@ -12,10 +12,8 @@ object Utils {
   val gen3Host = "data.kidsfirstdrc.org"
   val dcfHost = "api.gdc.cancer.gov"
   val patternUrnUniqueIdStudy = "[A-Z][a-z]+-(SD_[0-9A-Za-z]+)-([A-Z]{2}_[0-9A-Za-z]+)"
-  val documentReferenceExtract = "^DocumentReference\\/([0-9]+)$"
-  val specimenExtract = "^Specimen\\/([0-9]+)$"
   val phenotypeExtract = "^[A-Z]{2,}.[0-9]+$"
-
+  val DRS_HOSTNAME = "drs://data.kidsfirstdrc.org/"
```
Comment (evans-g-crsj) [inline import-task/src/main/scala/bio/ferlab/fhir/etl/Utils.scala:16]: https://github.com/kids-first/kf-task-fhir-etl/blob/ebebc60d439483ff161b70b507edf5bc07fd3d85/kf_task_fhir_etl/target_api_plugins/entity_builders/drs_document_reference.py#L372

File: import-task/src/main/scala/bio/ferlab/fhir/etl/Utils.scala [hunk 2]
```diff
@@ -12,10 +12,8 @@ object Utils {
   val gen3Host = "data.kidsfirstdrc.org"
   val dcfHost = "api.gdc.cancer.gov"
   val patternUrnUniqueIdStudy = "[A-Z][a-z]+-(SD_[0-9A-Za-z]+)-([A-Z]{2}_[0-9A-Za-z]+)"
-  val documentReferenceExtract = "^DocumentReference\\/([0-9]+)$"
-  val specimenExtract = "^Specimen\\/([0-9]+)$"
```
Comment (evans-g-crsj) [inline import-task/src/main/scala/bio/ferlab/fhir/etl/Utils.scala:16]: Unused

File: import-task/src/main/scala/bio/ferlab/fhir/etl/transformations/Transformations.scala [hunk 3]
```diff
@@ -240,41 +264,34 @@ object Transformations {
...
       grouped
+
     },
     Drop("type", "identifier", "collection", "subject",
       "parent",
-      "container", "collection_sample", "meta")
+      "container", "collection_sample")
   )
 
-  def probandObservationMappings(excludeCollection: Boolean): List[Transformation] = List(
```
Comment (evans-g-crsj) [inline import-task/src/main/scala/bio/ferlab/fhir/etl/transformations/Transformations.scala:277]: Unused

File: import-task/src/main/scala/bio/ferlab/fhir/etl/transformations/Transformations.scala [hunk 4]
```diff
@@ -226,12 +228,34 @@ object Transformations {
...
+      .withColumn("collection_sample", coalesce(parentRange.reverse.map(p => col(s"parent_$p")): _*))
+      .withColumn("collection_sample_id", noopWhenFlatModel(col("collection_sample.sample_id")))
+      .withColumn("collection_sample_type", noopWhenFlatModel(col("collection_sample.sample_type")))
+      .withColumn("collection_fhir_id", noopWhenFlatModel(col("collection_sample.fhir_id")))
+    val sampleWithParentFiltered =
+      if (isFlatSpecimenModel) samplesWithParent.where(col("collection_fhir_id") =!= col("fhir_id")) else samplesWithParent
+    sampleWithParentFiltered.drop(parentRange.map(p => s"parent_$p"): _*).select(struct(col("*")) as "specimen")
+  }
+
+  private def nullLitStr() = lit(null).cast("string")
```
Comment (jecos) [inline import-task/src/main/scala/bio/ferlab/fhir/etl/transformations/Transformations.scala:252]: could nbe a val

**Review by jecos — CHANGES_REQUESTED** [review]

**Review by jecos — APPROVED** [review]

**Pull request merged**